code/game/machinery/computer/access_terminal.dm

```diff
@@ -0,0 +1,103 @@
+
+/// Simple access terminal.
+/// Can put an ID in it, and set or unset some accesses.
+/// Intended to be used in ooc/admin spaces, like antag/actor prep areas.
+ABSTRACT_TYPE(/obj/machinery/computer/access_terminal)
+	name = "self-service access terminal"
+	desc = "A simple access terminal. It allows changing one's ID accesses."
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "altcomputerw"
+	light_color = LIGHT_COLOR_BLUE
+	icon_screen = "guest"
+	icon_scanline = "altcomputerw-scanline"
+	density = FALSE
+	appearance_flags = TILE_BOUND
+
+	var/obj/item/card/id/held_card
+
+/obj/machinery/computer/access_terminal/Destroy()
+	if (held_card)
+		held_card.forceMove(loc)
+		held_card = null
+	return ..()
+
+/// Should return a list of `/datum/access`.
+/obj/machinery/computer/access_terminal/proc/get_available_accesses()
+	. = list()
+
+/obj/machinery/computer/access_terminal/attackby(obj/item/attacking_item, mob/user)
+	var/obj/item/card/id/idcard = attacking_item
+	if(!held_card && istype(idcard))
+		usr.drop_from_inventory(idcard, src)
```

Comment:
Do not rely on `usr`, you have the mob passed in as a parameter (`user`)

Reply:
replaced with `user.`